pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -1020,9 +1021,9 @@ func (cb *ClusterBuilder) buildUpstreamClusterTLSContext(opts *buildClusterOpts,
 				ValidationContextSdsSecretConfig: authn_model.ConstructSdsSecretConfig(authn_model.SDSRootResourceName),
 			},
 		}
-		// Set default SNI of cluster name for istio_mutual if sni is not set.
+		// Set default SNI for istio_mutual if sni is not set.
 		if len(tlsContext.Sni) == 0 {
```

Comment:
For istio mtls we always set sni from `BuildDNSSrvSubsetKey`? So is this redundant

Reply:
That is what I am checking

Reply:
Looks like this is not needed